lib/python/picongpu/plugins/plot_mpl/energy_waterfall_visualizer.py

```diff
@@ -0,0 +1,169 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2018 PIConGPU contributors
+Authors: Sophie Rudat, Sebastian Starke
+License: GPLv3+
+"""
+
+from picongpu.plugins.data import EnergyHistogramData
+from picongpu.plugins.plot_mpl.base_visualizer import Visualizer as\
+    BaseVisualizer
+import numpy as np
+from matplotlib.colors import LogNorm
+import matplotlib.pyplot as plt
+
+
+class Visualizer(BaseVisualizer):
+    """
+    Class for creation of histogram plots on a logscaled y-axis.
+    """
+
+    def __init__(self, run_directories=None, ax=None):
+        """
+        Parameters
+        ----------
+        run_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        super().__init__(EnergyHistogramData, run_directories, ax)
+        self.cbar = None
+
+    def _create_plt_obj(self, idx):
+        """
+        Implementation of base class function.
+        Turns 'self.plt_obj' into a matplotlib.pyplot.plot object.
+        """
+        counts, bins, all_iterations, dt = self.data[idx]
+        np_data = np.zeros((len(bins), len(all_iterations)))
+        for index, ts in enumerate(all_iterations):
+            np_data[:, index] = counts[index]
+        ps = 1.e12  # for conversion from s to ps
+        max_iter = max(all_iterations * dt * ps)
+        self.plt_obj[idx] = self.ax.imshow(np_data, aspect="auto",
+                                           norm=LogNorm(),
+                                           origin="lower",
+                                           extent=(0, max_iter,
+                                                   0, max(bins*1.e-3)))
+        if self.cur_iteration:
+            self.plt_lin = self.ax.axvline(self.cur_iteration * dt * ps,
+                                           color='#FF6600')
+        self.cbar = plt.colorbar(self.plt_obj[idx], ax=self.ax)
```

Comment:
This might become problematic in the case of several plots since only one colorbar is attached. Shouldn't each `imshow` have its own colorbar? The same was necessary in the `phase_space_visualizer.py` file, you can have a look there on how to do this.